bin/deploy-aws

```diff
@@ -5,13 +5,33 @@ set -e;
 # abort on undefined variable
 set -u;
 
-# get current branch from git info
-PROJECT=${CI_PROJECT_NAME};
-BRANCH=$(echo "${CI_COMMIT_REF_NAME}" | cut -d'.' -f1-2); # first two parts, if tag name
+# CI_COMMIT_REF_NAME: The branch or tag name for which project is built
+# only the following options will be forwarded .gitlab-ci
+# - develop                 [for develop branch]
+# - vX.Y.Z                  [for stable releases]
+# - vX.Y.Z-(alpha|beta|rc)? [for pre-releases]
+BRANCHORTAG=${CI_COMMIT_REF_NAME};
+MINOR=$(echo "${BRANCHORTAG}" | cut -s -d'.' -f1-2); # vX.Y
+MAJOR=$(echo "${BRANCHORTAG}" | cut -s -d'.' -f1); # vX
+PRERELEASE=$(echo ${BRANCHORTAG} | cut -s -d'-' -f2); # null or (alpha|beta|rc)?
 
 # set project base
 SOURCE_PATH="build";
-DEPLOY_PATH="vf";
+DEPLOY_BASE="vf";
+
+# set deploy paths
+if [ "${BRANCHORTAG}" = "develop" ]; then
+  # if this is the develop branch, only deploy to develop folder
+  DEPLOY_PATHS="${DEPLOY_BASE}/${BRANCHORTAG}";
+elif [ -n "${PRERELEASE}" ]; then
+  # if this is a pre-release, only deploy to pre-release folder
+  DEPLOY_PATHS="${DEPLOY_BASE}/${BRANCHORTAG}";
```

Comment:
Deploy Path need to be PRERELEASE?

Reply:
no ${BRANCHORTAG} is the whole tag
${PRERELEASE} is just the alpha|beta|rc bit and I check if it is a non-zero-length string for the purposes to checking if this is a pre-release